src/notifications/Manager.cpp

```diff
@@ -0,0 +1,90 @@
+#include "notifications/Manager.h"
+
+#include "Cache.h"
+#include "EventAccessors.h"
+#include "Logging.h"
+#include "MatrixClient.h"
+#include "Utils.h"
+
+#include <QFile>
+#include <QImage>
+#include <QStandardPaths>
+
+#include <mtxclient/crypto/client.hpp>
+
+QString
+NotificationsManager::cacheImage(const mtx::events::collections::TimelineEvents &event)
+{
+        const auto url      = mtx::accessors::url(event);
+        auto encryptionInfo = mtx::accessors::file(event);
+
+        auto filename = QString::fromStdString(mtx::accessors::body(event));
+        QString path{QStandardPaths::writableLocation(QStandardPaths::CacheLocation) + "/" +
+                     filename};
```

Comment:
The name should include the size to avoid duplicates and stuff overwriting each other.